### ISSUE TITLE
Upgrade to Nuxt 3.8 causes "Failed to execute 'insertBefore' on 'Node': The node before which the new node is to be inserted is not a child of this node."

### ISSUE THREAD

**Issue by jonatanvm** [nuxt/nuxt#24045]
### Environment

- Operating System: Linux
- Node Version:     v20.8.0
- Nuxt Version:     3.8.0
- CLI Version:      3.9.1
- Nitro Version:    2.7.2
- Package Manager:  npm@10.1.0
- Builder:          -
- User Config:      devtools, css, modules, postcss, runtimeConfig, app, routeRules, security, typescript
- Runtime Modules:  @pinia/nuxt@0.5.1, nuxt-security@1.0.0-rc.2
- Build Modules:    -

### Reproduction

N/A
### Describe the bug

After upgrading to Nuxt 3.8 I get "Failed to execute 'insertBefore' on 'Node'" on each first load of my app. 

### Additional context

Happens regardless of if I soft upgrade or hard upgrade using nuxi upgrade.

### Logs

```shell-script
Uncaught (in promise) DOMException: Failed to execute 'insertBefore' on 'Node': The node before which the new node is to be inserted is not a child of this node.
    at insert (http://localhost:5173/_nuxt/node_modules/.cache/vite/client/deps/chunk-UJJZ7LWV.js?v=4fc7bb9c:9174:12)
    at processFragment (http://localhost:5173/_nuxt/node_modules/.cache/vite/client/deps/chunk-UJJZ7LWV.js?v=4fc7bb9c:6926:7)
    at patch (http://localhost:5173/_nuxt/node_modules/.cache/vite/client/deps/chunk-UJJZ7LWV.js?v=4fc7bb9c:6441:9)
    at ReactiveEffect.componentUpdateFn [as fn] (http://localhost:5173/_nuxt/node_modules/.cache/vite/client/deps/chunk-UJJZ7LWV.js?v=4fc7bb9c:7202:9)
    at ReactiveEffect.run (http://localhost:5173/_nuxt/node_modules/.cache/vite/client/deps/chunk-UJJZ7LWV.js?v=4fc7bb9c:422:19)
    at instance.update (http://localhost:5173/_nuxt/node_modules/.cache/vite/client/deps/chunk-UJJZ7LWV.js?v=4fc7bb9c:7243:52)
    at callWithErrorHandling (http://localhost:5173/_nuxt/node_modules/.cache/vite/client/deps/chunk-UJJZ7LWV.js?v=4fc7bb9c:1552:32)
    at flushJobs (http://localhost:5173/_nuxt/node_modules/.cache/vite/client/deps/chunk-UJJZ7LWV.js?v=4fc7bb9c:1755:9)
    at flushJobs (http://localhost:5173/_nuxt/node_modules/.cache/vite/client/deps/chunk-UJJZ7LWV.js?v=4fc7bb9c:1765:7)
insert	@	chunk-UJJZ7LWV.js:9174
processFragment	@	chunk-UJJZ7LWV.js:6926
patch	@	chunk-UJJZ7LWV.js:6441
componentUpdateFn	@	chunk-UJJZ7LWV.js:7202
run	@	chunk-UJJZ7LWV.js:422
instance.update	@	chunk-UJJZ7LWV.js:7243
callWithErrorHandling	@	chunk-UJJZ7LWV.js:1552
flushJobs	@	chunk-UJJZ7LWV.js:1755
flushJobs	@	chunk-UJJZ7LWV.js:1765
Promise.then (async)		
nextTick	@	chunk-UJJZ7LWV.js:1635
onResolve	@	nuxt-layout.js?v=4fc7bb9c:46
triggerEvent	@	chunk-UJJZ7LWV.js:2494
resolve	@	chunk-UJJZ7LWV.js:2811
(anonymous)	@	chunk-UJJZ7LWV.js:2906
Promise.then (async)		
registerDep	@	chunk-UJJZ7LWV.js:2870
mountComponent	@	chunk-UJJZ7LWV.js:7034
processComponent	@	chunk-UJJZ7LWV.js:6994
patch	@	chunk-UJJZ7LWV.js:6467
mountChildren	@	chunk-UJJZ7LWV.js:6711
processFragment	@	chunk-UJJZ7LWV.js:6928
patch	@	chunk-UJJZ7LWV.js:6441
componentUpdateFn	@	chunk-UJJZ7LWV.js:7137
run	@	chunk-UJJZ7LWV.js:422
instance.update	@	chunk-UJJZ7LWV.js:7243
setupRenderEffect	@	chunk-UJJZ7LWV.js:7251
mountComponent	@	chunk-UJJZ7LWV.js:7041
processComponent	@	chunk-UJJZ7LWV.js:6994
patch	@	chunk-UJJZ7LWV.js:6467
mountSuspense	@	chunk-UJJZ7LWV.js:2515
process	@	chunk-UJJZ7LWV.js:2461
patch	@	chunk-UJJZ7LWV.js:6492
componentUpdateFn	@	chunk-UJJZ7LWV.js:7137
run	@	chunk-UJJZ7LWV.js:422
instance.update	@	chunk-UJJZ7LWV.js:7243
setupRenderEffect	@	chunk-UJJZ7LWV.js:7251
mountComponent	@	chunk-UJJZ7LWV.js:7041
processComponent	@	chunk-UJJZ7LWV.js:6994
patch	@	chunk-UJJZ7LWV.js:6467
mountChildren	@	chunk-UJJZ7LWV.js:6711
mountElement	@	chunk-UJJZ7LWV.js:6618
processElement	@	chunk-UJJZ7LWV.js:6583
patch	@	chunk-UJJZ7LWV.js:6455
componentUpdateFn	@	chunk-UJJZ7LWV.js:7137
run	@	chunk-UJJZ7LWV.js:422
instance.update	@	chunk-UJJZ7LWV.js:7243
setupRenderEffect	@	chunk-UJJZ7LWV.js:7251
(anonymous)	@	chunk-UJJZ7LWV.js:2884
Promise.then (async)		
registerDep	@	chunk-UJJZ7LWV.js:2870
mountComponent	@	chunk-UJJZ7LWV.js:7034
processComponent	@	chunk-UJJZ7LWV.js:6994
patch	@	chunk-UJJZ7LWV.js:6467
mountSuspense	@	chunk-UJJZ7LWV.js:2515
process	@	chunk-UJJZ7LWV.js:2461
patch	@	chunk-UJJZ7LWV.js:6492
componentUpdateFn	@	chunk-UJJZ7LWV.js:7137
run	@	chunk-UJJZ7LWV.js:422
instance.update	@	chunk-UJJZ7LWV.js:7243
setupRenderEffect	@	chunk-UJJZ7LWV.js:7251
mountComponent	@	chunk-UJJZ7LWV.js:7041
processComponent	@	chunk-UJJZ7LWV.js:6994
patch	@	chunk-UJJZ7LWV.js:6467
render2	@	chunk-UJJZ7LWV.js:7761
mount	@	chunk-UJJZ7LWV.js:5206
app.mount	@	chunk-UJJZ7LWV.js:10592
initApp	@	entry.js:55
await in initApp (async)		
(anonymous)	@	entry.js:64



Uncaught (in promise) DOMException: Failed to execute 'insertBefore' on 'Node': The node before which the new node is to be inserted is not a child of this node.
    at insert (http://localhost:5173/_nuxt/node_modules/.cache/vite/client/deps/chunk-UJJZ7LWV.js?v=4fc7bb9c:9174:12)
    at processFragment (http://localhost:5173/_nuxt/node_modules/.cache/vite/client/deps/chunk-UJJZ7LWV.js?v=4fc7bb9c:6926:7)
    at patch (http://localhost:5173/_nuxt/node_modules/.cache/vite/client/deps/chunk-UJJZ7LWV.js?v=4fc7bb9c:6441:9)
    at ReactiveEffect.componentUpdateFn [as fn] (http://localhost:5173/_nuxt/node_modules/.cache/vite/client/deps/chunk-UJJZ7LWV.js?v=4fc7bb9c:7202:9)
    at ReactiveEffect.run (http://localhost:5173/_nuxt/node_modules/.cache/vite/client/deps/chunk-UJJZ7LWV.js?v=4fc7bb9c:422:19)
    at instance.update (http://localhost:5173/_nuxt/node_modules/.cache/vite/client/deps/chunk-UJJZ7LWV.js?v=4fc7bb9c:7243:52)
    at callWithErrorHandling (http://localhost:5173/_nuxt/node_modules/.cache/vite/client/deps/chunk-UJJZ7LWV.js?v=4fc7bb9c:1552:32)
    at flushJobs (http://localhost:5173/_nuxt/node_modules/.cache/vite/client/deps/chunk-UJJZ7LWV.js?v=4fc7bb9c:1755:9)
    at flushJobs (http://localhost:5173/_nuxt/node_modules/.cache/vite/client/deps/chunk-UJJZ7LWV.js?v=4fc7bb9c:1765:7)
```

**Comment by jeannen** [comment]
Same here

**Comment by huang-julien** [comment]
Hello :wave: your repo is private. Are you using `.client` components ? If so, a fix will be released for it in 3.8.1

**Comment by layen520** [comment]
same problem
use client in my router middleware

**Comment by jonatanvm** [comment]
> Hello 👋 your repo is private. Are you using `.client` components ? If so, a fix will be released for it in 3.8.1

Yes, using `.client` components! When is the estimated release? Will the problem resolve if i remove `.client` and wrap the components in `<client-only>`?

**Comment by layen520** [comment]
I am use process.client in router middleware. Is there a problem?

**Comment by dargmuesli** [comment]
Fyi, nuxt 3.8.1 has been released.

**Comment by danielroe** [comment]
I believe this issue has likely been resolved in Vue 3.4. If not, please provide a reproduction and I will happily reopen.
